### PR TITLE
[#951] Extend the Domainmodel Example by hyperlinking in javadoc.

### PR DESCRIPTION
**Pull request by miklossy**
- Modify the DomainmodelHyperlinkHelper to recognize hyperlinks in
javadocs.
- Implement corresponding HyperlinkingTest test cases.

Signed-off-by: Tamas Miklossy <miklossy@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/navigation/DomainmodelHyperlinkHelper.java [hunk 1]
```diff
@@ -84,6 +91,16 @@ public void createHyperlinksByOffset(XtextResource resource, int offset, IHyperl
 				}
 			}
 		}
+		
+		createHyperlinksInJavadoc(resource, offset, acceptor);
 	}
 
+	private void createHyperlinksInJavadoc(XtextResource resource, int offset, IHyperlinkAcceptor acceptor) {
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/navigation/DomainmodelHyperlinkHelper.java]: i wonder if we should make this available via injectable boolean in xbase
@szarnekow what do you think?

Comment (szarnekow) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/navigation/DomainmodelHyperlinkHelper.java]: Or we make this available by default and opt-out would be to override the method?

Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/navigation/DomainmodelHyperlinkHelper.java]: :+1:

Comment (miklossy) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/navigation/DomainmodelHyperlinkHelper.java]: Actually I wonder why the `DomainmodelHyperlinkHelper` inherits from `TypeAwareHyperlinkHelper` and not from `XbaseHyperLinkHelper`. The `XbaseHyperLinkHelper` class is already aware of [createHyperlinksInJavaDoc](https://github.com/eclipse/xtext-eclipse/blob/2027a7e48acafb161bc3ee2d861a3b3e36cc8d6d/org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/navigation/XbaseHyperLinkHelper.java#L162).

Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/navigation/DomainmodelHyperlinkHelper.java]: XbaseHyperLinkHelper seems way newer than TypeAwareHyperlinkHelper. thus this seens to be a migration issue when XbaseHyperLinkHelper was introduced

Comment (miklossy) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/navigation/DomainmodelHyperlinkHelper.java]: Then simply changing the inheritance would sufficient, isnt it?

Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/navigation/DomainmodelHyperlinkHelper.java]: Assume so

Comment (miklossy) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/navigation/DomainmodelHyperlinkHelper.java]: @szarnekow What do you think?

Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/navigation/DomainmodelHyperlinkHelper.java]: ping @szarnekow

Comment (szarnekow) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/navigation/DomainmodelHyperlinkHelper.java]: True. It certainly looks like that would be the right thing to do.

**Comment by miklossy** [comment]
Thanks for the feedback, I updated the PR as requested.

**Pull request merged**